src/sagemaker_xgboost_container/data_utils.py

```diff
@@ -300,15 +300,17 @@ def _get_csv_dmatrix_file_mode(files_path, csv_weights):
     return dmatrix
 
 
-def _get_csv_dmatrix_pipe_mode(pipe_path, csv_weights):
+def _get_csv_dmatrix_pipe_mode(pipe_path, csv_weights, subsample_on_read):
```

Comment:
1. Should we rename to indicate a ratio? I read this right now as a boolean. 
2. Should it have a default of `1` i.e. no subsampling?

Reply:
Agree with the first point.

The second point is debateable because if the value is None then it the code flow doesn't even go through the subsampling section. However, I'm also not fond of overloading arguments (ie, None in one case, and then ratio in another. Is this pythonic best practices?).

We can also have default be 1, and the check for subsampling does:

```
if submsampling is < 1 and subsampling > 0:
    do stuff
```

And this way we don't need a None default.

Reply:
1. Yeah I can rename it to `subsample_ratio_on_read`.
2. This is an internal function. The default value (`None`) is set in `get_csv_dmatrix`.

Reply:
If you think of `subsample_ratio_on_read` as an optional parameter, having a value of `None` vs. `1` has a semantic difference. As an API user with `None` you explicitly specify that you do not want to have any subsampling; with `1`, although the result will be the same as if `None` is specified, you do not know for sure whether there will be some kind of additional logic executed related to subsampling.

Reply:
FYI: XGBoost has a similar parameter (https://xgboost.readthedocs.io/en/latest/parameter.html#parameters-for-tree-booster) that has default value `1`. 

Q: Under what circumstance would an API user specify `subsample_ratio_on_read=None`? 
- If no subsampling is desired then user could just not specify the parameter. 
- Additionally if `subsample_ratio_on_read=1` then behavior *should* be same as `None`. 

So it's not clear to me when the `None` parameter should be used.

Reply:
I changed the accepted range of values to (0, 1); meaning, if specified, it has to be greater than zero and less than 1. If `None`, then no subsampling is performed. I still think that `None` is more distinct than `1` to indicate that no subsampling should be performed.